spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/AbstractConnectionFactoryConfigurer.java

```diff
@@ -0,0 +1,61 @@
+package org.springframework.boot.autoconfigure.amqp;
+
+import org.springframework.amqp.rabbit.connection.AbstractConnectionFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionNameStrategy;
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.util.Assert;
+
+/**
+ * Configures {@link AbstractConnectionFactory Rabbit ConnectionFactory} with sensible
+ * defaults.
+ *
+ * @param <T> the connection factory type.
+ * @author Chris Bono
+ * @since 2.6.0
+ */
+public abstract class AbstractConnectionFactoryConfigurer<T extends AbstractConnectionFactory> {
```

Comment:
While SB only auto-configures AbstractConnectionFactory impl of CachingConnectionFactory currently, I went ahead and split it out into ABC + impl specific concrete class to set the specific properties. This would allow some other consumer to extend the ABC for say PooledConnectionFactory or ThreadConnectionFactory.